src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -473,7 +474,183 @@ pass::EliminateSqueeze::EliminateSqueeze() {
         return false;
     };
 
-    auto m = std::make_shared<pattern::Matcher>(squeeze_pattern, matcher_name);
+    auto m = make_shared<pattern::Matcher>(squeeze_pattern, matcher_name);
+    this->register_matcher(m, callback);
+}
+
+namespace {
+bool check_squeeze(const shared_ptr<Node>& node) {
+    auto squeeze = dynamic_pointer_cast<ov::opset9::Squeeze>(node);
+    if (squeeze) {
+        auto axis = dynamic_pointer_cast<ov::opset9::Constant>(squeeze->input_value(1).get_node_shared_ptr());
+        if (axis) {
+            auto axis_val = axis->cast_vector<int64_t>();
+            if (axis_val.size() == 1 && axis_val[0] == 1) {
```

Comment:
is that ok that we don't allow negative axis for Squeeze?

Reply:
fixed